### PR TITLE
♻️ Make DN404 zksync / abstract compatible

### PR DESCRIPTION
**Pull request by Vectorized**
## Description

♻️
 
## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [x] Ran `forge fmt`?
- [x] Ran `forge snapshot`?
- [x] Ran `forge test`?

_Pull requests with an incomplete checklist will be thrown out._

### CODE REVIEW

File: src/DN404.sol [hunk 1]
```diff
@@ -129,6 +129,11 @@ abstract contract DN404 {
     /// [Etherscan](https://etherscan.io/address/0x000000000022D473030F116dDEE9F6B43aC78BA3)
     address internal constant _PERMIT2 = 0x000000000022D473030F116dDEE9F6B43aC78BA3;
 
+    /// @dev The ZKsync Permit2 deployment.
+    /// If deploying on ZKsync or Abstract, override `_isPermit2(address)` to check against this too.
+    /// [Etherscan](https://era.zksync.network/address/0x0000000000225e31D15943971F47aD3022F714Fa)
+    address internal constant _ZKSYNC_PERMIT_2 = 0x0000000000225e31D15943971F47aD3022F714Fa;
```
Comment (cygaar) [inline src/DN404.sol:135]: The abstract permit2 address is actually: 0x6b4420f66De496D18A6c36367cf7f1440fd9289f

Do you need it specifically at the address above?

File: src/DN420.sol [hunk 2]
```diff
@@ -177,6 +177,11 @@ abstract contract DN420 {
     /// [Etherscan](https://etherscan.io/address/0x000000000022D473030F116dDEE9F6B43aC78BA3)
     address internal constant _PERMIT2 = 0x000000000022D473030F116dDEE9F6B43aC78BA3;
 
+    /// @dev The ZKsync Permit2 deployment.
+    /// If deploying on ZKsync or Abstract, override `_isPermit2(address)` to check against this too.
+    /// [Etherscan](https://era.zksync.network/address/0x0000000000225e31D15943971F47aD3022F714Fa)
+    address internal constant _ZKSYNC_PERMIT_2 = 0x0000000000225e31D15943971F47aD3022F714Fa;
```
Comment (cygaar) [inline src/DN420.sol:183]: Same comment as above

**Review by cygaar — COMMENTED** [review]
lgtm outside of the permit2 address

File: src/DN404.sol (hunk 1 above)
Comment (Vectorized) [reply src/DN404.sol:135]: Oh my. fragmentation. Best to get it at the same address!

Comment (Vectorized) [reply src/DN404.sol:135]: It’s not yet up, https://explorer.testnet.abs.xyz/address/0x6b4420f66De496D18A6c36367cf7f1440fd9289f

so better fix now b4 mainnet.

Comment (cygaar) [reply src/DN404.sol:135]: I'll try to get it deployed to the same address

**Comment by Vectorized** [comment]
Much thanks @cygaar @coffeexcoin

**Pull request merged**